connect-button/src/main/java/com/ifttt/connect/ui/ConnectButton.java

```diff
@@ -252,6 +268,8 @@ public boolean onPreDraw() {
     protected void onAttachedToWindow() {
         super.onAttachedToWindow();
         lifecycleRegistry.markState(Lifecycle.State.STARTED);
+
+        if (!analyticsOptOut) { analyticsManager.trackUiImpression(AnalyticsObject.fromConnnection(connectButton.getConnection()), AnalyticsLocation.fromConnectButton(getContext())); }
```

Comment:
I think we should do all of the tracking within BaseConnectButton, so that all of them are contained within one place.

This is also not correct: when the view is attached to window, we don't necessarily have the connection data yet.

Reply:
yes, this was something I forgot to remove.